chore: Add focus management documentation (#1274)
Amsterdam/design-system: storybook/src/docs/usability.docs.mdx

{/* @license CC0-1.0 */}

import { Meta } from "@storybook/blocks";

<Meta title="Docs/Design Guidelines/Usability" />

# Usability

## Focus management

> In traditional multi-page applications (MPAs) moving to another page means resetting focus to the `<body>` element,

Reviewer: ```suggestion
In traditional multi-page applications (MPAs), navigating to another page resets the focus to the `<body>` element,
```

or the first element with an `autofocus` attribute if there is one.
For single-page applications (SPAs) this isn’t always done consistently.

> For example, Next.js sets the focus on a new page at roughly the same position as it was on the old page,
> resulting in an unexpected focus position.

Reviewer: ```suggestion
For example, Next.js sets the focus on a new page at roughly the same position as it was on the old page.
This can result in an unexpected focus position, e.g. when navigating between pages of a form.
```

Reviewer 2: ```suggestion
For example, Next.js sets the focus on a new page at roughly the same position as it was on the old page.
This can cause the focus to change unexpectedly, like when you move between pages in a form.
```

Author: > This can cause the focus to change unexpectedly, like when you move between pages in a form.

This kind of changes the meaning of that sentence.

Make sure to correct this focus behaviour in your application.

> The simplest and most consistent way to do this is to set focus on the `<body>` element when moving to another page.

Reviewer: ```suggestion
The simplest and most consistent way to do this is to set focus on the `<body>` element after navigating to another page.
```

Or ‘when loading a page’?

Reviewer 2: "after navigating" seems good to me.

> For some applications it might make more sense to set the focus somewhere else when moving to a new page,

Reviewer: ```suggestion
For some applications it might make more sense to set the focus somewhere else,
```

but be very careful with this.

> Your assumptions on how your users use your app might not be correct.

Reviewer 2: ```suggestion
Your assumptions on how users use the app might not be correct.
```

Too many you's.
